### PR TITLE
[host] Copy hostfxr and hostpolicy for Mono also, if !mobile

### PR DESCRIPTION
**Pull request by lambdageek**
Fixes the mono "HelloWorld" sample (src/mono/samples/HelloWorld) on desktop platforms

### CODE REVIEW

**Review by directhex — APPROVED** [review]

**Review by ViktorHofer — CHANGES_REQUESTED** [review]
Let's wait for @agocke's review as he just recently made the same change just a bit differently.

File: src/libraries/externals.csproj [hunk 1]
```diff
@@ -95,6 +95,9 @@
           AfterTargets="AfterResolveReferences"
           Condition="'$(RuntimeFlavor)' == 'Mono'">
     <ItemGroup>
+      <!-- copy the host files to make the desktop samples work -->
+      <RuntimeFiles Include="@(HostFxrFile)" Condition="Exists('@(HostFxrFile)') and '$(TargetsMobile)' != 'true'"/>
+      <RuntimeFiles Include="@(HostPolicyFile)" Condition="Exists('@(HostPolicyFile)') and '$(TargetsMobile)' != 'true'" />
```
[the remark this answers is not in the document]
Comment (lambdageek) [reply src/libraries/externals.csproj:100]: that's for the test host.

The change in this PR is for the "runtime pack" that is used by `eng/targetingpacks.targets`

**Comment by lambdageek** [comment]
> Let's wait for @agocke's review as he just recently made the same change just a bit differently.

Andy is OOF for a few months.

This PR is addressing the changes from #73095 that broke mono desktop samples

**Review by ViktorHofer — APPROVED** [review]
Just saw that the same already exists for CoreClr in the `OverrideRuntimeMono` target so it should be OK.

**Pull request merged**